SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskProcessBuilder.java

```diff
@@ -63,9 +63,9 @@ public ProcessBuilder call() throws Exception {
     if (task.getTaskInfo().hasContainer() && task.getTaskInfo().getContainer().hasDocker()) {
       executorUtils.sendStatusUpdate(task.getDriver(), task.getTaskInfo(), TaskState.TASK_STARTING, String.format("Pulling image... (executor pid: %s)", executorPid), task.getLog());
       try {
-        dockerClient.pull(task.getTaskInfo().getContainer().getDocker().getImage());
-      } catch (UncheckedTimeoutException te) {
-        throw new ProcessFailedException(String.format("Timed out trying to reach docker daemon after %s seconds", configuration.getDockerClientTimeLimitSeconds()));
+        dockerUtils.pull(task.getTaskInfo().getContainer().getDocker().getImage());
+      } catch (DockerException e) {
+        throw new ProcessFailedException(String.format("Could not pull docker image due to error: %s", e));
```

Comment:
Don't add the exception to an error message, include the exception as a cause argument.